pkg/kubelet/status/status_manager.go

```diff
@@ -466,8 +464,26 @@ func (m *manager) TerminatePod(pod *v1.Pod) {
 		}
 	}
 
+	// Make sure all pods are transitioned to a terminal phase.
+	// TODO(#116484): Also assign terminal phase to static an pods.
+	if !kubetypes.IsStaticPod(pod) {
+		switch status.Phase {
+		case v1.PodSucceeded, v1.PodFailed:
+			// do nothing, already terminal
+		case v1.PodPending:
+			klog.V(3).InfoS("Marking terminal pending pod as failed", "phase", status.Phase, "pod", klog.KObj(pod), "podUID", pod.UID)
+			status.Phase = v1.PodFailed
+		case v1.PodRunning:
+			klog.InfoS("Terminal running pod should have already been marked as failed, programmer error", "pod", klog.KObj(pod), "podUID", pod.UID)
```

Comment:
Pinning the discussion here https://github.com/kubernetes/kubernetes/pull/115331#issuecomment-1472461616. This might actually be logged if the pod is deleted from API server. I have reproduced it locally too. I think we should either log programmer error only when the pod status is not in `podStatuses`, or just log it in a neutral form (as for pending pods).

Reply:
Went on for logging the programmer error if that status is still cached. PTAL.